nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -817,4 +823,25 @@ Costmap2DROS::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameter
   return result;
 }
 
+void Costmap2DROS::getCostCallback(
+  const std::shared_ptr<rmw_request_id_t>,
+  const std::shared_ptr<nav2_msgs::srv::GetCost::Request> request,
```

Comment:
Request: It was made by someone in the original ticket that we add the option to use the `footprint` for the cost instead of the point-cost if the `GetCost` request field `use_footprint` was set to `true` (can be default `false`)

Reply:
Added use_footprint bool in `GetCost` srv